#VRML_SIM R2022a utf8
WorldInfo {
  info [
    "The humanoid sprint benchmark aims at programing a NAO robot in Python to run a 10 meter race. "
    "The goal is to run the 10 meter as fast as possible. "
    "All the sensors and actuators of the robot can be used to achieve the best performance. "
    "Records are stored on the challenge web site. "
  ]
  title "Humanoid Sprint"
  window "humanoid_sprint"
  basicTimeStep 20
  FPS 20
  contactProperties [
    ContactProperties {
      material1 "NAO foot material"
      coulombFriction [
        7
      ]
      bounce 0.3
      bounceVelocity 0.003
    }
  ]
}
Viewpoint {
  orientation -0.0453775182199013 -0.017577934865303874 0.9988152467028493 3.880000989054137
  position -2.872861838972636 2.262796184115301 0.7908396211803081
}
TexturedBackground {
  texture "noon_cloudy_countryside"
}
TexturedBackgroundLight {
  texture "noon_cloudy_countryside"
  luminosity 0.4
  castShadows FALSE
}
SpotLight {
  attenuation 0 0 1
  beamWidth 0.2
  cutOffAngle 0.7
  intensity 10
  location -5 -0.5 4
  castShadows TRUE
}
SpotLight {
  attenuation 0 0 1
  beamWidth 0.2
  cutOffAngle 0.7
  intensity 10
  location 0 -0.5 4
  castShadows TRUE
}
SpotLight {
  attenuation 0 0 1
  beamWidth 0.2
  cutOffAngle 0.7
  intensity 10
  location 5 -0.5 4
  castShadows TRUE
}
DEF TRACK Floor {
  translation 0 -0.39 0
  size 11 3
  tileSize 11 3
  appearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/samples/robotbenchmark/humanoid_sprint/worlds/textures/nao_10m_track_base_color.jpg"
      ]
    }
    roughness 1
    roughnessMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/samples/robotbenchmark/humanoid_sprint/worlds/textures/nao_10m_track_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/samples/robotbenchmark/humanoid_sprint/worlds/textures/nao_10m_track_normal.jpg"
      ]
    }
  }
}
DEF GRASS Solid {
  translation 0 0 -0.005
  children [
    Shape {
      appearance Grass {
        colorOverride 0.905486 0.988983 0.47155
        textureTransform TextureTransform {
          scale 20 20
        }
      }
      geometry Cylinder {
        bottom FALSE
        height 0.001
        radius 10
        side FALSE
      }
    }
  ]
  name "grass"
  model "floor"
  locked TRUE
}
DEF Robot Robot {
  translation -5.21 -0.14 0.1
  rotation 0 1 0 0
  children [
    DEF armright Hinge2Joint {
      jointParameters HingeJointParameters {
        anchor -1.73472e-18 -0.05 0.04
      }
      jointParameters2 JointParameters {
      }
      device [
        RotationalMotor {
          name "shoulderright"
        }
      ]
      device2 [
        PositionSensor {
          name "posright"
        }
      ]
      endPoint Solid {
        translation 4.171665427066013e-17 -0.17000000000000007 0.0399999
        rotation 1 -3.915708159101289e-17 -4.607956604547773e-16 1.570799999999999
        children [
          DEF armright Shape {
            appearance PBRAppearance {
            }
            geometry Capsule {
              height 0.2
              radius 0.01
            }
          }
        ]
        name "solid(1)"
        boundingObject USE armright
        physics Physics {
        }
      }
    }
    DEF armleft Hinge2Joint {
      jointParameters HingeJointParameters {
        axis 0 5 0
        anchor 0 0.05 0.05
      }
      jointParameters2 JointParameters {
      }
      device [
        RotationalMotor {
          name "shoulderleft"
        }
      ]
      device2 [
        PositionSensor {
          name "posleft"
        }
      ]
      endPoint Solid {
        translation -7.041358955504738e-18 0.1700000000000001 0.03999999999999999
        rotation -0.9999999999999999 -7.840024969030136e-17 7.794442199847306e-17 1.5707953071795855
        children [
          DEF armlaft Shape {
            appearance PBRAppearance {
            }
            geometry Capsule {
              height 0.2
              radius 0.01
            }
          }
        ]
        boundingObject USE armlaft
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation -0.05 0 -0.07
      rotation 0 0 1 3.14159
      children [
        DEF back_sensor Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dsback"
      boundingObject USE back_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0 -0.05 -0.07
      rotation 0 0 1 -1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dsright"
      boundingObject Shape {
        appearance PBRAppearance {
        }
        geometry Box {
          size 0.01 0.01 0.01
        }
      }
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0 0.05 -0.07
      rotation 0 0 1 1.5708
      children [
        DEF left_sensor Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dsleft"
      boundingObject USE left_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.05 0.05 -0.07
      rotation 0 0 1 0.523599
      children [
        DEF front_left_sensor Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dsfrontright"
      boundingObject USE front_left_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.0473205 -0.05 -0.07
      rotation 0 0 1 -0.523595307179586
      children [
        DEF front_right_sensor Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dsfrontleft"
      boundingObject USE front_right_sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.06 0 -0.07
      rotation 0 1 0 0
      children [
        DEF sensorbody Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "dsfrontmiddle"
      boundingObject USE sensorbody
      physics Physics {
      }
    }
    DEF RoboterBody Shape {
      appearance PBRAppearance {
        baseColor 0.5 1 0.4
      }
      geometry Box {
        size 0.1 0.1 0.2
      }
    }
  ]
  name "My_Robo"
  boundingObject USE RoboterBody
  physics Physics {
  }
  controller "walking"
}
Robot {
  name "supervisor"
  controller "supervisor_controller"
  supervisor TRUE
}
